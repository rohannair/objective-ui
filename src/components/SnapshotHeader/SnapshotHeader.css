@import '../../styles/globals';

.objectiveEditor {
  &:hover {
    .editButton {
      display: inline-block;
      visibility: visible;
      cursor: pointer
    }
  }
}

.editButton {
  color: #616161;
  visibility: hidden;
  font-size: 85%;
  height: 1rem;
  margin-left: 0.3em;

  &:hover {
    color: #009ED9;
  }

  &:active {
    position: relative;
    top: 1px;
   }
}

.addButton {
	span {
  color: #777;
  vertical-align: middle;
  line-height: 1;
  font-size: 62%;
  background-color: transparent;
  border: dashed;
  border-color: #ccc;
  border-width: 2px;
  padding: 2px 4px;
	}

  &:hover {
    cursor: pointer;

    span {
      cursor: pointer;
      color: #009ED9;
      border-color: #009ED9;
    }
  }

  &:active {
    position: relative;
    top: 1px;
  }
}

