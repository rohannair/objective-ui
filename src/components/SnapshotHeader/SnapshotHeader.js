import React, { PropTypes } from 'react'
import gql from 'graphql-tag'
import styled from 'styled-components'

import Pill from '../Pill'
import UserTab from '../UserTab'
import ObjectiveTab from '../ObjectiveTab'

const SnapshotHeader = styled((snap) => (
  <div className={snap.className}>
    <UserTab {...snap.user} >
      <ObjectiveTab
      readOnly={snap.readOnly}
      objective={snap.objective}
      editObjective={snap.editObjective}
      blocker={snap.blocker}
      createdAt={snap.createdAt}
      />
    </UserTab>
  </div>
  ))`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  padding: 0 10px;
`

SnapshotHeader.mutations = {
  editSnapshotObjective : gql`
    mutation editSnapshotObjective($objectiveId: String, $id: Int!) {
      editSnapshotObjective(objectiveId: $objectiveId, id: $id) {
        id
        objective {
          name
        }
      }
    }
  `
}

SnapshotHeader.fragments = {
  header: gql`
    fragment SnapshotHeaderFragment on Snapshot {
      createdAt
      blocker
      objective {
        name
      }
      user {
        id
        ...UserTabFragment
      }
    }
    ${UserTab.fragments.user}
  `
}

export default SnapshotHeader
