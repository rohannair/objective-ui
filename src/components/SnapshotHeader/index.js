import SnapshotHeader from './SnapshotHeader'
export default SnapshotHeader
