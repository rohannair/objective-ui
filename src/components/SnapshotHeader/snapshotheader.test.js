import React from 'react'

import SnapshotHeader from './index.js'

describe('Test suite for SnapshotHeader component', () => {
  it('SnapshotHeader should exist', () => {
    let wrapper = shallow(<SnapshotHeader/>)
    expect(wrapper).toMatchSnapshot()
  })
})

