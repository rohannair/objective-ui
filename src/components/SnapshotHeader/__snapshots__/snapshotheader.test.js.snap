exports[`Test suite for SnapshotHeader component SnapshotHeader should exist 1`] = `
<Component
  className="jHuiZP" />
`;
